pil-analyzer/src/pil_analyzer.rs

```diff
@@ -317,9 +322,54 @@ impl PILAnalyzer {
         }
     }
 
+    pub fn traits_resolution(&mut self) {
+        fn collect_references(expr: &mut Expression) -> Vec<&mut Reference> {
+            match expr {
+                Expression::Reference(_, ref mut r @ Reference::Poly(_)) => vec![r],
+                _ => expr.children_mut().flat_map(collect_references).collect(),
+            }
+        }
+
+        let traits_functions_defs: HashMap<String, FunctionValueDefinition> = self
+            .definitions
+            .iter()
+            .filter_map(|(name, (_, def))| match def {
+                Some(FunctionValueDefinition::TraitFunction(_, _)) => {
+                    Some((name.clone(), def.clone().unwrap()))
```

Comment:
Please don't clone, especially not the def

Reply:
Actually why do you need this hashmap any way? Can't you just use `self.definitions`?